tests/app/test_utils.py

```diff
@@ -51,3 +55,43 @@ def test_get_midnight_for_day_before_returns_expected_date(date, expected_date):
 def test_midnight_n_days_ago(current_time, arg, expected_datetime):
     with freeze_time(current_time):
         assert midnight_n_days_ago(arg) == expected_datetime
+
+
+@freeze_time('2019-01-10 00:30')
+def test_get_notification_table_to_use(sample_service):
+    # it's currently early morning of Thurs 10th Jan.
+    # When the delete task runs a bit later, it'll delete data from last wednesday 2nd.
+    assert get_notification_table_to_use(sample_service, 'sms', date(2018, 12, 31), False) == NotificationHistory
+    assert get_notification_table_to_use(sample_service, 'sms', date(2019, 1, 1), False) == NotificationHistory
+    assert get_notification_table_to_use(sample_service, 'sms', date(2019, 1, 2), False) == Notification
+    assert get_notification_table_to_use(sample_service, 'sms', date(2019, 1, 3), False) == Notification
+
+
+@freeze_time('2019-01-10 00:30')
+def test_get_notification_table_to_use_knows_if_delete_task_has_run(sample_service):
+    # it's currently early morning of Thurs 10th Jan.
+    # The delete task deletes/moves data from last wednesday 2nd.
+    assert get_notification_table_to_use(sample_service, 'sms', date(2019, 1, 2), False) == Notification
+    assert get_notification_table_to_use(sample_service, 'sms', date(2019, 1, 2), True) == NotificationHistory
+
+
+@freeze_time('2019-06-09 23:30')
+def test_get_notification_table_to_use_respects_daylight_savings_time(sample_service):
+    # current time is 12:30am on 10th july in BST
+    assert get_notification_table_to_use(sample_service, 'sms', date(2019, 6, 1), False) == NotificationHistory
+    assert get_notification_table_to_use(sample_service, 'sms', date(2019, 6, 2), False) == Notification
+
+
+@freeze_time('2019-01-10 00:30')
+def test_get_notification_table_to_use_checks_service_data_retention(sample_service):
+    create_service_data_retention(sample_service, 'email', days_of_retention=3)
```

Comment:
Do we need a test for a service with data retention longer than the default 7 days?

Reply:
added to that test